File: Getting-Practice-with-Python-and-Javascript/5-Object Oriented Programming/Javascript/Excercises/program-6.js
// 6. **Clase Cuenta Bancaria**: Crea una clase CuentaBancaria con atributos para el titular, el saldo, y el tipo de cuenta. Incluye métodos para depositar, retirar, y obtener la información de la cuenta.


class CuentaBancaria {
    constructor(titular, saldo, tipoCuenta) {
        this.titular = titular
        this.saldo = saldo
        this.tipoCuenta = tipoCuenta
    }

    verInfo() {
        let resultado = `Informacion de la cuenta: \n`
        resultado += `Titular: ${this.titular} \n`
        resultado += `Saldo: $${this.saldo} \n`
        resultado += `Tipo de Cuenta: ${this.tipoCuenta} \n`
        return resultado;
    }

    depositar() {
        console.log(`Saldo actual en su Cuenta... ${this.saldo}`)
        let deposito = parseFloat(prompt('Ingrese el monto a depositar: '))
        if (deposito > 0 && !isNaN(deposito)) {
            this.saldo += deposito
            return `Usted ah depositado Exitosamente $${deposito}, en su cuenta Bancaria!`
        } else {
            return `El monto a depositar debe ser Positivo.`
        }
    }

    retirar() {
        console.log(`Saldo disponible en su cuenta: $${this.saldo}`)
        let retiro = parseFloat(prompt('Ingrese un monto a retirar: '))
        if (this.saldo >= retiro && !isNaN(retiro)) {
            this.saldo -= retiro
            return `Usted ah retirado con exito $${retiro} de su cuenta Bancaria!`;
        } else {
            return `Saldo insuficiente...`;
        }
    }
}

let cliente1 = new CuentaBancaria('Imanol Aguer', 66485, 'Caja De Ahorro')


// Ver info de la Cuenta
console.log(cliente1.verInfo())


// Depositar
console.log(cliente1.depositar())
console.log(cliente1.verInfo())

// Retirar
console.log(cliente1.retirar())
console.log(cliente1.verInfo())

File: Getting-Practice-with-Python-and-Javascript/5-Object Oriented Programming/Python/Excercises/program-6.py
# 6. **Clase Cuenta Bancaria**: Crea una clase CuentaBancaria con atributos para el titular, el saldo, y el tipo de cuenta. Incluye métodos para depositar, retirar, y obtener la información de la cuenta.

class CuentaBancaria:
    def __init__(self, titular, saldo, tipoCuenta):
        self.titular = titular
        self.saldo = saldo
        self.tipoCuenta = tipoCuenta

    def __str__(self):
        resultado = f'Informacion de la Cuenta: \n'
        resultado += f'Titular: {self.titular} \n'
        resultado += f'Saldo: {self.saldo} \n'
        resultado += f'Tipo de Cuenta: {self.tipoCuenta} \n'
        return resultado

    def depositar(self):
        print(f'Saldo actual en su cuenta: ... {self.saldo}')
        deposito = float(input('Ingrese el monto a depositar: '))
        if deposito > 0:
            self.saldo += deposito
            print(
                f'Usted ah depositado excitosamente ${deposito} a su cuenta bancaria.')
        else:
            print(
                f'El monto a depositar debe ser positivo.')

    def retirar(self):
        print(f'Saldo disponible para retirar ...{self.saldo}')
        retiro = float(input('Ingrese el monto a retirar de la ucenta: '))
        if self.saldo >= retiro:
            self.saldo -= retiro
            print(
                f'Usted ah retirado excitosamente ${retiro} de su cuenta bancaria.')
        else:
            print(
                f'Saldo insuficiente... el monto a retirar supera el saldo disponible en su cuenta')


# TEST
cliente1 = CuentaBancaria('Imanol', 66485, 'Caja de Ahorro')

# ver informacion de la cuenta
print(cliente1)

# depositar
cliente1.depositar()
print(cliente1)

# retirar
cliente1.retirar()
print(cliente1)
